src/qtt/qiskit/passes.py

```diff
@@ -4,21 +4,44 @@
 import numpy as np
 import qiskit
 from qiskit.circuit import Barrier, Delay, Reset
-from qiskit.circuit.library import (CRXGate, CRYGate, CRZGate, CZGate,
-                                    PhaseGate, RXGate, RYGate, RZGate, U1Gate,
+from qiskit.circuit.library import (CRXGate, CRYGate, CRZGate, CZGate, PhaseGate, RXGate, RYGate, RZGate, U1Gate,
                                     U2Gate, U3Gate, UGate)
-from qiskit.circuit.library.standard_gates import (CU1Gate, RZZGate, SdgGate,
-                                                   SGate, TdgGate, TGate,
-                                                   ZGate)
+from qiskit.circuit.library.standard_gates import CU1Gate, RZZGate, SdgGate, SGate, TdgGate, TGate, ZGate
 from qiskit.circuit.quantumcircuit import QuantumCircuit
 from qiskit.converters.circuit_to_dag import circuit_to_dag
 from qiskit.dagcircuit import DAGCircuit
-from qiskit.dagcircuit.dagnode import DAGInNode, DAGNode, DAGOpNode, DAGOutNode
+from qiskit.dagcircuit.dagnode import DAGInNode, DAGOpNode
 from qiskit.transpiler.basepasses import TransformationPass
 
 logger = logging.getLogger(__name__)
 
 
+class RemoveGateByName(TransformationPass):
+    """Return a circuit with all gates with specified name removed.
```

Comment:
Just out of curiosity: What is the use case for this pass?

Reply:
The qiskit algorithms generate circuits with have identity gates in them. The QI interface has no identity gate, we we remove them before passing to the Spin-2 backend. @QFer